add in privacy page

joelwass opened the pull request:
Added in consent checkbox to signup flow

<img width="1146" alt="Screen Shot 2020-11-23 at 1 31 21 PM" src="https://user-images.githubusercontent.com/7344422/100017714-33a62880-2d90-11eb-9012-e2b62970d00a.png">

joelwass committed:
add in privacy page

joelwass committed:
update

stripedpajamas commented on components/common/header/authNav.js:
```diff
@@ -111,6 +112,20 @@ const AuthNav = ({ user, onLogout }) => {
+                <a>
+                  <Box as='span' display='flex' alignItems='center'>
+                    <LockIcon marginRight='.5rem' />
+                    <span>Privacy</span>
```
I don't think it needs to go in the header, just the footer.

stripedpajamas commented on components/common/header/header.js:
```diff
@@ -49,6 +49,13 @@ const Header = () => {
+                <TextLink
+                  display={['none', 'none', 'none', 'inline']}
+                  href='/privacy'
+                  text='Privacy'
```
should be Privacy Policy

stripedpajamas commented on pages/privacy.js:
```diff
@@ -0,0 +1,254 @@
+        <Divider />
+        <FAQHeading>Privacy Policy</FAQHeading>
+        <PrivacyTextBlock>
+          Our Privacy Policy was posted on 20 November 2020 and last updated on 20 November 2020. It governs the privacy terms of our website, located at https://flossbank.com, and the tools we provide you (the "Website" or the "Service").
```
```suggestion
          Our Privacy Policy was posted on 20 November 2020 and last updated on 20 November 2020. It governs the privacy terms of our website, located at https://flossbank.com, and the tools we provide (the "Website" or the "Service").
```

Taking off the "you" since the parenthetical becomes unclear with it present.

stripedpajamas commented on pages/privacy.js:
```diff
@@ -0,0 +1,254 @@
+    >
+      <UnderlinedHeading as='h1' text='Privacy Policy' align='center' />
+      <Card shadowSz='lg' maxW='60rem'>
+        <FAQHeading>TLDR</FAQHeading>
+        <PrivacyTextBlock>
+          We know privacy policies can be riddled with heavy language and legalese that’s difficult to parse. Our TLDR is that we will not share your personal information no matter what.
+        </PrivacyTextBlock>
+        <PrivacyTextBlock fontWeight='500'>
+          The only information we have is your email address (used for authentication), your GitHub Id (used to see if you’re a GitHub admin for an organization you’re trying to donate on behalf of) and the OSS packages you’ve installed (used to distribute your donation / ad revenue).
+        </PrivacyTextBlock>
+        <PrivacyTextBlock fontWeight='500'>
+          That's it
+        </PrivacyTextBlock>
+        <PrivacyTextBlock>
+          Please accept the following privacy policy as the official language of what’s stated above.
+        </PrivacyTextBlock>
```
I don't really like the TLDR, I would delete it or make it a one liner like "We will never share your personal data."

stripedpajamas commented on pages/privacy.js:
```diff
@@ -0,0 +1,254 @@
+          <ListItem>
+          Tracking &amp; Cookies Data <br />
+We use cookies to hold certain information during your session on our website.
+Cookies are files with small amount of data which may include an anonymous unique identifier. Cookies are sent to your browser from a website and stored on your device.<br />
```
```suggestion
Cookies are files with small amounts of data which may include a unique identifier. Cookies are sent to your browser from The Service and stored on your device.<br />
```

stripedpajamas commented on pages/privacy.js:
```diff
@@ -0,0 +1,254 @@
+          </ListItem>
+          <ListItem>
+          Usage Data <br />
+We may also collect information how the Service is accessed and used ("Usage Data"). This Usage Data contains which OSS packages were installed by you, at what time, and what ads were seen.
```
```suggestion
We may also collect information how the Service is accessed and used ("Usage Data"). This Usage Data contains the names and versions of packages installed using the Flossbank CLI, the name and version of the package manager used to install the packages, the time of the installations, and which ads were seen during the installations (if any).
```

stripedpajamas commented on pages/privacy.js:
```diff
@@ -0,0 +1,254 @@
+          Business Transaction. If we are involved in a merger, acquisition or asset sale, your Personal Data may be transferred. We will provide notice before your Personal Data is transferred and becomes subject to a different Privacy Policy.
+          </ListItem>
+          <ListItem>
+          Disclosure for Law Enforcement. Under certain circumstances, we may be required to disclose your Personal Data if required to do so by law or in response to valid requests by public authorities (e.g. a court or a government agency).
```
would love this to not be the case. are you set on this

joelwass replied:
this is required by law, unsure if we can avoid this. if it's required by law, it needs to be in the privacy policy to let users know

joelwass replied:
changed to just "if required by law"

stripedpajamas commented on pages/privacy.js:
```diff
@@ -0,0 +1,254 @@
+        Your consent to this Privacy Policy followed by your submission of such information represents your agreement to that transfer.
+        </PrivacyTextBlock>
+        <Divider />
+        <FAQHeading>Disclosure Of Data</FAQHeading>
```
don't these 2 exceptions fly in the face of the TLDR?

we will never share your data... unless someone buys us... or the government asks us.

joelwass replied:
i don't think they do. we won't share your data. if we're bought, your data will be transferred and if we're required by law to turn over our db's, we will not break the law. 

i think we can never avoid having to share data if required by law or if we're bought. that being said, i think that's all the more reason to keep the data we store to the bare minimum, which is what we're doing.

stripedpajamas commented on pages/privacy.js:
```diff
@@ -0,0 +1,254 @@
+          The right to data portability. You have the right to be provided with a copy of your Personal Data in a structured, machine-readable and commonly used format.
+          </ListItem>
+          <ListItem>
+          The right to withdraw consent. You also have the right to withdraw your consent at any time where Flossbank relied on your consent to process your personal information.
```
This implies there was consent to begin with... so we will need to add something on the sign up page like a check box or text that says they are consenting. 

Also because we don't use cookies before sign up, I think (hoping) we can avoid the GDPR pop up on the splash page and instead have GDPR consent baked into the sign up page or something.

joelwass replied:
I think this line is saying "if we rely on your consent to process any personal information, you can always revoke that consent"

we should also have a checkbox that points them to the privacy policy i think

stripedpajamas replied:
can you include the checkbox in this PR

joelwass replied:
done

stripedpajamas commented on pages/privacy.js:
```diff
@@ -0,0 +1,254 @@
+        <FAQHeading>Service Providers</FAQHeading>
+        <PrivacyTextBlock>We may employ third party companies and individuals to facilitate our Service ("Service Providers"), to provide the Service on our behalf, to perform Service-related services or to assist us in analyzing how our Service is used.</PrivacyTextBlock>
+        <PrivacyTextBlock>These third parties have access to your Personal Data only to perform these tasks on our behalf and are obligated not to disclose or use it for any other purpose.</PrivacyTextBlock>
+        <PrivacyTextBlock>Analytics</PrivacyTextBlock>
```
this and "Payments" below don't look good on the page-- maybe bold them or something.

joelwass committed:
Update pages/privacy.js

Co-authored-by: Peter Squicciarini <stripedpajamas273@gmail.com>

joelwass committed:
Update pages/privacy.js

Co-authored-by: Peter Squicciarini <stripedpajamas273@gmail.com>

joelwass committed:
Update pages/privacy.js

Co-authored-by: Peter Squicciarini <stripedpajamas273@gmail.com>

joelwass committed:
updated privacy policy

stripedpajamas commented:
![image](https://user-images.githubusercontent.com/2707340/99999356-e10b4300-2d74-11eb-8e5c-c79cdd1aa65e.png)

Latest deployment link still shows analytics/payments sections looking pretty rough on my side.

joelwass committed:
added checkbox and updated

joelwass commented:
Cleaned it up and added to the TLDR. without bullet points i feel like the TLDR is worthless. as a user i'd want to go and quickly scan and say ok, makes sense, sgtm

joelwass committed:
remove console

joelwass committed:
fix
